jdaviz/core/template_mixin.py

```diff
@@ -1031,8 +1031,12 @@ def selected_subset_state(self):
 
     @property
     def selected_subset_mask(self):
-        get_data_kwargs = {'data_label': self.plugin.dataset.selected,
-                           'subset_to_apply': self.selected}
+        if self._allowed_type == 'spatial':
+            get_data_kwargs = {'data_label': self.plugin.dataset.selected,
+                               'spatial_subset': self.selected}
+        elif self._allowed_type == 'spectral':
+            get_data_kwargs = {'data_label': self.plugin.dataset.selected,
+                               'spectral_subset': self.selected}
```

Comment:
Is this too cryptic?  Or maybe could cause issues if we ever generalize `_allowed_type` but not `get_data`? 🤷 

```suggestion
        get_data_kwargs = {'data_label': self.plugin.dataset.selected,
                           f'{self._allowed_type}_subset': self.selected}
```

Reply:
This works for now but your point about generalizing `_allowed_type` still stands.